torch/distributed/distributed_c10d.py

```diff
@@ -573,11 +573,29 @@ class GroupMember(metaclass=_WorldMeta):
 STORE_BASED_BARRIER_PREFIX = "store_based_barrier_key"
 
 
-def _get_object_coll_device(group: Optional[ProcessGroup] = None):
+def _get_pg_default_device(group: Optional[ProcessGroup] = None):
+    """
+    Returns the device to use with ``group`` for object collectives and barrier.
+    There are selection rules:
+        1. If user specifies exactly one backend in ``init_process_group`` call:
+            use that backend
+        2. Else if user specifies multiple "device:backend" pairs in init_process_group:
+            If "cpu" is among those pairs, use "cpu" (because the object is in cpu memory);
+            Otherwise, use the first backend (sort of a random pick).
+
+    Args:
+        group (ProcessGroup, optional): The process group to work on. If None,
+            the default process group will be used.
+
+    Returns:
+        A distributed request object.
```

Comment:
nit: maybe a typo? It should return a device instead of a request object.
But thanks for adding the documentation here.